### PR TITLE
Fixes bug where raw text update overwrites all text

### PR DESCRIPTION
**Pull request by rozele**
After fixing issues with textTransform, 278a3c13 introduced a bug where updates to a raw text node can overwrite the TextBlock parent if it has only a single Span child inline. This change ensures we check that the updated raw text Run is indeed the direct child of a TextBlock.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/7572)

### CODE REVIEW

**Comment by rozele** [comment]
Before:

https://user-images.githubusercontent.com/1106239/114481529-bd854f80-9bd2-11eb-9379-b2204a3edd1f.mp4

After:

https://user-images.githubusercontent.com/1106239/114481578-ddb50e80-9bd2-11eb-99d0-3e08514f9e03.mp4

File: packages/@react-native-windows/tester/src/js/examples-win/Text/TextExample.windows.tsx [hunk 1]
```diff
@@ -197,9 +197,11 @@ export class TextExample extends React.Component<
               </Text>
             </Text>
             <Text onPress={() => this.setState({toggle2: !this.state.toggle2})}>
-              Click to change raw text:{' '}
-              <Text style={{textTransform: 'uppercase'}}>
-                Hello {this.state.toggle2 ? 'Earth' : 'World'}
+              <Text>
```
Comment (rectified95) [inline packages/@react-native-windows/tester/src/js/examples-win/Text/TextExample.windows.tsx:200]: Why do you need this new nested Text tag here?

Comment (rozele) [reply packages/@react-native-windows/tester/src/js/examples-win/Text/TextExample.windows.tsx:200]: We don't, adding it allowed me to create the repro above. It doesn't hurt to keep it there, but we can revert it if you'd like.

File: vnext/Microsoft.ReactNative/Views/RawTextViewManager.cpp [hunk 2]
```diff
@@ -90,7 +91,9 @@ void RawTextViewManager::NotifyAncestorsTextChanged(ShadowNodeBase *nodeToUpdate
       } else if (!std::wcscmp(nodeType, L"RCTVirtualText") && textTransform == TextTransform::Undefined) {
         textTransform = static_cast<VirtualTextShadowNode *>(parent)->textTransform;
       }
+
       parent = static_cast<ShadowNodeBase *>(host->FindShadowNodeForTag(parent->GetParent()));
+      isNested = true;
```
Comment (rectified95) [inline vnext/Microsoft.ReactNative/Views/RawTextViewManager.cpp:96]: We seem to always set this after the while loop - what does this variable track exactly?

File: packages/@react-native-windows/tester/src/js/examples-win/Text/TextExample.windows.tsx (hunk 1 above)
Comment (rectified95) [reply packages/@react-native-windows/tester/src/js/examples-win/Text/TextExample.windows.tsx:200]: I was thinking more of keeping both scenarios - if the new one helped catch this edge case, it looks likely the old one might do that as well in the future.

File: vnext/Microsoft.ReactNative/Views/RawTextViewManager.cpp (hunk 2 above)
Comment (rozele) [reply vnext/Microsoft.ReactNative/Views/RawTextViewManager.cpp:96]: It tracks whether the Run that's being updated is the direct child of the TextBlock, in which case its value is "lifted" into the Text property to enable text optimization. It's intended to be true after the first pass through the loop.

Comment (rectified95) [reply vnext/Microsoft.ReactNative/Views/RawTextViewManager.cpp:96]: Sounds good

File: packages/@react-native-windows/tester/src/js/examples-win/Text/TextExample.windows.tsx (hunk 1 above)
Comment (rectified95) [reply packages/@react-native-windows/tester/src/js/examples-win/Text/TextExample.windows.tsx:200]: Unless that's covered in another example already.

**Review by rectified95 — APPROVED** [review]

File: packages/@react-native-windows/tester/src/js/examples-win/Text/TextExample.windows.tsx (hunk 1 above)
Comment (rozele) [reply packages/@react-native-windows/tester/src/js/examples-win/Text/TextExample.windows.tsx:200]: There should be more than enough coverage from the other examples.

**Pull request merged**